htlcswitch/link.go

```diff
@@ -2812,6 +2827,25 @@ func (l *channelLink) processRemoteSettleFails(fwdPkg *channeldb.FwdPkg,
 				continue
 			}
 
+			// If the failure message lacks an HMAC (but includes
+			// the 4 bytes for encoding the message and padding
+			// lengths, then this means that we received it as an
+			// UpdateFailMalformedHTLC. As a result, we'll signal
+			// that we need to convert this error within the switch
+			// to an actual error, by encrypting it as if we were
+			// the originating hop.
+			convertedErrorSize := lnwire.FailureMessageLength + 4
+			convertedError := len(pd.FailReason) ==
+				convertedErrorSize
+
+			// If the fail reason length is at least the length of
+			// the variable length marker, clip off the marker
+			// bytes.
+			failReason := pd.FailReason
+			if len(pd.FailReason) >= varLenFailureMarkerLen {
+				failReason = failReason[varLenFailureMarkerLen:]
```

Comment:
I think this logic would also have to be in `reforwardSettleFails` in the switch. otherwise i think it's possible that we receive a failure reason near 65535 bytes, add 300 bytes, and then the switch reforwards the failure and the incoming link is unable to send the message given the wire constraint

Reply:
Ah yes, it should indeed. But I guess this PR is no longer relevant as we decided to only allow 256+ byte failures.